_posts/en/newsletters/2020-01-08-newsletter.md

```diff
@@ -0,0 +1,95 @@
+  relevant after reports that a popular exchange was investigating users
+  participating in the chaumian-style coinjoins created by Wasabi
+  Wallet.  Although several ideas were discussed, we think the summary
+  by Lucas Ontivero captures the essence of the overall conclusion: "In
+  summary, unequal inputs/outputs coinjoins [knapsack][] is the best we
+  have at the moment ([but] it is not as effective as equal-outputs
+  transactions)."
+
+- **Encoded descriptors:** Chris Belcher [asked][belcher descriptors]
+  for feedback from the Bitcoin-Dev mailing list about base64 encoding
+  [output script descriptors][topic descriptors] so that they're easier
+  to copy and paste (and also so that regular users aren't exposed to
+  their code-like syntax).  At least one reply was opposed to the idea
+  and other replies which did support the idea each proposed a different
+  encoding format.  The discussion to date did not come to any
+  clear conclusion.
+
+## Notable code and documentation changes
+
+*Notable changes this week in [Bitcoin Core][bitcoin core repo],
+[C-Lightning][c-lightning repo], [Eclair][eclair repo], [LND][lnd repo],
+[libsecp256k1][libsecp256k1 repo], [Bitcoin Improvement Proposals
+(BIPs)][bips repo], and [Lightning BOLTs][bolts repo].*
+
+- [C-Lightning #3351][] extends the `invoice` RPC with a new
+  `exposeprivatechannels` parameter than allows the user to request
+  the addition of route hints for private channels to a generated [BOLT11][]
+  invoice.  The user may optionally specific which channels they want to
+  advertise in the invoice, including both public and private channels.
+
+- [LND #3657][] switches from using base64 to hex for the display of
+  various binary data.  This is a breaking API change for users of any
+  of the updated fields.
+
+- [LND #3814][] allows the the UTXO sweeper to add wallet inputs to a
+  sweep transaction in order to ensure its output meets the dust limit.
+  This is designed to help support the proposed anchor outputs feature
+  (see [Newsletter #70][news70 anchor]) which will need to add inputs to
+  a transaction in order to be able to spend low-value UTXOs.
+
+{% include linkers/issues.md issues="3657,3814,3351" %}
+[wuille slides]: https://prezi.com/view/AlXd19INd3isgt3SvW8g/
+[wuille suggestion]: http://www.erisian.com.au/meetbot/taproot-bip-review/2019/taproot-bip-review.2019-12-17-19.01.log.html#l-8
+[zmn post]: https://lists.linuxfoundation.org/pipermail/lightning-dev/2019-December/002375.html
+[ficsor non-equal]: https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2019-December/017538.html
+[cashfusion]: https://github.com/cashshuffle/spec/blob/master/CASHFUSION.md
+[knapsack]: https://www.comsys.rwth-aachen.de/fileadmin/papers/2017/2017-maurer-trustcom-coinjoin.pdf
+[coinjoin sudoku]: http://www.coinjoinsudoku.com/
+[payment batching]: https://github.com/bitcoinops/scaling-book/blob/master/x.payment_batching/payment_batching.md
```

Comment:
Note that this link is likely to change (it's currently chapter x but will be updated when we add more content to the book)

Reply:
Yeah, if we want stable links, we should probably remove numbers from the chapter URLs.  I think we briefly discussed this before, but I'm still in favor of eventually moving the book content to the website---that will give us stable URLs (and the ability to create redirect links if the URLs aren't as stable as predicted), integration with our current QA/CI, better formatting than GitHub (e.g. native footnote support), and CSS customization.

Reply:
> I'm still in favor of eventually moving the book content to the website

I agree that makes sense. Do you want to do that now or when we have a bit more content?

Reply:
How about we do after reviewing the next proposed chapter in the other repository but before we announce the publication of that chapter; that way we can bundle the move announcement with the new-chapter announcement.

Reply:
SGTM!